bazel/astore/defs.bzl

```diff
@@ -188,19 +193,32 @@ astore_download = rule(
           mandatory = False,
           default = "",
         ),
-        "_astore_client": attr.label(
-            default = Label("//astore/client:astore"),
+        # This needs to be overriden if using the local enkit binary
+        # since the target //astore/client:astore will still build from source
+        # even if it is not used since it is declared as a dependency.
+        "astore_client": attr.label(
             allow_single_file = True,
             executable = True,
             cfg = "host",
         ),
+        "local": attr.bool(
+            doc = "Use the local enkit binary to perform astore download operations.",
+            default = False,
+        ),
     },
     doc = """Downloads artifacts from artifact store - astore.
 
 With this rule, you can easily download
 files from an artifact store.""",
 )
 
+def astore_download(local = False, *args, **kwargs):
+    if local:
+        kwargs["local"] = local
+    else:
+        kwargs["astore_client"] = Label("//astore/client:astore")
+    return _astore_download(*args, **kwargs)
+
```

Comment:
Is the macro specifically necessary, if _astore_client is not added to the dependencies? If not, it can be dropped

Reply:
I modified the macro in a subsequent commit, but the macro is still required since the default behavior is to build astore from source which is the primary use-case. However for PD and DFT targets that need to opt-out of the default usage pattern of astore_download, the bazel rule needs to be explicitly told to omit the astore dependency. This is to avoid having to update every instance of astore_download in the repo to explicitly declare `astore_client = Label("//astore/client:astore")`.